server/map_rpc_server.go

```diff
@@ -217,6 +232,11 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 			req.MapId,
 			writeRev,
 			hasher, func(ctx context.Context, f func(context.Context, storage.MapTreeTX) error) error {
+				if t.opts.UseSingleTransaction {
+					glog.V(1).Infof("EXP: using single tx")
+					txRolledUp++
```

Comment:
Is this thread-safe?  (Won't the various `subtreeWriter.buildSubtree` goroutines all be accessing this in parallel?)

Reply:
Yeah, I was assuming 64bit increment was atomic (might be on x64 I guess, but it's naughty).
Fixed.